src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -86,8 +89,12 @@ internal FeedAndEntryMaterializerAdapter(ODataFormat odataFormat, ODataReaderWra
             if (isDeltaPayload)
             {
                 // If we have a delta payload, we need the stack data structure to keep track of the read items.
-                this.materializerStateItems = new Stack<IMaterializerState>();
-            }    
+                this.deltaMaterializerStateItems = new Stack<IMaterializerState>();
+            }
+            else
+            {

```

Comment:
will this be created all the time even when we are not doing a deep insert? Is it possible to check whether we are doing a deep insert or not from this point?

Reply:
Is it possible to use the same `stack` for both deep updates and deep inserts instead of having different `stacks` of the same type?

Reply:
We can't use the same stack since in `DeltaSet` requests, we validate that the top-level object is not a `MaterializerEntry`.